src/backend/joanie/core/serializers/client.py

```diff
@@ -66,49 +88,63 @@ def validate(self, attrs):
         """
         Check access rights specific to writing (create/update)
         """
-        username = self.context["request"].user.username
-
-        if not models.CourseAccess.objects.filter(
-            course=self.context["course_id"],
-            user__username=username,
-            role__in=[enums.OWNER, enums.ADMIN],
-        ).exists():
-            raise exceptions.PermissionDenied(
-                _(
-                    "You must be administrator or owner of a course to manage its accesses."
+        request = self.context["request"]
```

Comment:
If we need to have a request, I feel this logic should at the viewset level 🤔

Reply:
I tried to see how it can be ported to the viewset but it's not trivial since the viewset delegates very earlier to the serializer :thinking:  Since this pattern was already their before this PR, I propose to leave it like this and think about it for a further PR.